compiler/rustc_middle/src/mir/interpret/allocation/init_mask.rs

```diff
@@ -64,6 +64,13 @@ impl InitMask {
         }
     }
 
+    pub fn get_blocks(&self) -> Box<std::slice::Iter<'_, u64>> {
+        match &self.blocks {
+            InitMaskBlocks::Lazy { state: _ } => Box::new([].iter()),
```

Comment:
this is wrong, it's not zero sized, but repeating `state` for the length of the entire `bytes` buffer.

But, as Ralf mentioned, we can avoid this entirely by encoding `bytes` as `Vec<Option<u8>>` and using https://github.com/RalfJung/minirust/blob/196da9640a4cd608405ad40110d29930827a99a5/tooling/minimize/src/constant.rs#L132-L145 to generate that `Vec`. Then we don't need an `init_mask` field at all.